lib/agent/providers/hardware/index.js

```diff
@@ -105,17 +107,20 @@ exp.track_hardware_changes = (data) => {
     }
   }
 
-  device_keys.exist('hardware', (err, stored_data) => {
+  storage.do('query', {type: 'keys', column: 'id', data: 'hardware'}, (err, stored_data) => {
     if (err) logger.error('Unable to read hardware data');
-    if (!stored_data) return save_data();
+    if (stored_data.length == 0) return save_data();
 
-    stored_data = stored_data[0];
+    try {
+      stored_data = JSON.parse(stored_data[0].value);
+    } catch (e) {
+      console.log("ERROR!")  // modificar
```

Comment:
Y algo así no sería más útil?
```suggestion
      console.log(e)
```